shared/route-tree/render-route.js

```diff
@@ -33,10 +34,10 @@ export type RouteProps<P, S> = {
   shouldRender: boolean,
 
   // Route props (query params)
-  routeProps: P,
+  routeProps: KBRecord<P>, // Really a Map but typed as a record
```

Comment:
dunno if there's a better way to do this. Basically we pass in an object as the definition, and the implementation is a map of this type (so its a record) but its not actually backed by a record. I think its effectively the same to lie about it

cc: @MarcoPolo